src/sentry/eventstream/base.py

```diff
@@ -22,6 +32,16 @@ class ForwarderNotRequired(NotImplementedError):
     """
 
 
+class GroupState(TypedDict):
+    is_new: bool
+    is_regression: bool
+    is_new_group_environment: bool
+    primary_hash: Optional[str]
```

Comment:
How come it's optional? Shouldn't any group have a manatory primary hash?

Reply:
We don't really need the primary hash for performance issues. It's only really used here https://github.com/getsentry/sentry/blob/1c5976ce038f60219d50a04647bb4a4128184ccf/src/sentry/tasks/post_process.py#L501-L507, and actually, all of the handlers just ignore it, so maybe we should just remove this altogether.